src/storage/SparqlResourceStore.ts

```diff
@@ -0,0 +1,753 @@
+              ],
+            },
+            {
+              type: 'graph',
+              name: namedNode(parentContainerURI),
+              triples: [
+                quad(namedNode(parentContainerURI), CONTAINS_PREDICATE, namedNode(containerURI)),
+              ],
+            },
+          ],
+          insert: [],
+          where: [
+            {
+              type: 'bgp',
+              triples: [
+                quad(variable('s'), variable('p'), variable('o')),
+              ],
+            },
+          ],
+        },
+      ],
+      type: 'update',
+      prefixes: {},
+    } as Update;
+    return this.sendSparqlUpdate(deleteContainerQuery);
+  }
+
+  /**
+   * Checks whether the specified container is empty.
+   * Ignores the .metadata file corresponding to the container.
+   * @param containerURI - Identifier of the container.
+   */
+  private async isEmptyContainer(containerURI: string): Promise<boolean> {
+    const containerQuery = {
+      queryType: 'ASK',
+      where: [
+        {
+          type: 'graph',
+          name: namedNode(containerURI),
+          triples: [
+            quad(namedNode(containerURI), CONTAINS_PREDICATE, variable('o')),
+            {
+              type: 'filter',
+              expression: {
+                type: 'operation',
+                operator: '!=',
+                args: [
+                  variable('o'),
+                  namedNode(`${containerURI}.metadata`),
+                ],
+              },
+            },
+          ],
+        },
+      ],
+      type: 'query',
+    } as unknown as AskQuery;
+    const result = await this.sendSparqlQuery(containerQuery);
+    return !result.boolean;
+  }
+
+  /**
+   * Helper function without extra validation to get all triples in a graph corresponding to the specified URI.
+   * @param URI - URI of the resource.
+   */
+  private async getSparqlRepresentation(URI: string): Promise<any> {
+    const representationQuery = {
+      queryType: 'CONSTRUCT',
+      template: [
+        quad(variable('s'), variable('p'), variable('o')),
+      ],
+      where: [
+        {
+          type: 'graph',
+          name: namedNode(URI),
+          patterns: [
+            {
+              type: 'bgp',
+              triples: [
+                quad(variable('s'), variable('p'), variable('o')),
+              ],
+            },
+          ],
+        } as GraphPattern,
+      ],
+      type: 'query',
+      prefixes: {},
+    } as ConstructQuery;
+    return (await this.sendSparqlQuery(representationQuery)).results.bindings;
+  }
+
+  /**
+   * Helper function to get the representation of a document resource.
+   * @param resourceURI - Identifier of the resource to retrieve.
+   */
+  private async getResourceRepresentation(resourceURI: string): Promise<Representation> {
+    // Get the triples from the resourceURI graph and from the corresponding metadata graph.
+    const data: Quad[] = await this.getSparqlRepresentation(resourceURI);
```

Comment:
It would be great if you could make sure there is no `await` here. I would prefer `getSparqlRepresentation` to return a stream of data. This way that stream could be put directly into the Representation object without having to fully read it and in case large datasets get streamed they don't have to be fully kept in memory. Probably that the direct output will need to be piped into a Transform stream, but it should still stay streaming.

For metadata it will still be necessary to fully read the data, but that is never going to be a big dataset.